plot_data/core.py

```diff
@@ -1302,6 +1325,41 @@ def __init__(self, plots: List[PlotDataObject], sizes: List[Window] = None, elem
         PlotDataObject.__init__(self, type_='multiplot', name=name)
 
 
+def to_html_file(plot_data_object: PlotDataObject, debug_mode: bool = False, canvas_id: str = 'canvas',
+                 force_version: str = None, width: int = 750, height: int = 400, page_name: str = None):
+
+    template = plot_data_object.template
+
+    if force_version is not None:
+        version, folder, filename = get_current_link(version=force_version)
+    else:
+        version, folder, filename = get_current_link()
+    cdn_url = 'https://cdn.dessia.tech/js/plot-data/{}/{}'
+    lib_path = cdn_url.format(version, filename)
+    if debug_mode:
+        core_path = os.sep.join(os.getcwd().split(os.sep)[:-1] + [folder, filename])
+
+        if not os.path.isfile(core_path):
+            msg = 'Local compiled {} not found, fall back to CDN'
+            print(msg.format(core_path))
+        else:
+            lib_path = core_path.replace(" ", "%20")
+
+    s = template.substitute(data=json.dumps(plot_data_object.to_dict()), core_path=lib_path,
+                            canvas_id=canvas_id, width=width, height=height)
+    if not page_name:
+        temp_file = tempfile.mkstemp(suffix='.html')[1]
+        with open(temp_file, 'wb') as file:
+            file.write(s.encode('utf-8'))
+        file_name = temp_file
+    else:
+        with open(page_name + '.html', 'wb') as file:
+            file.write(s.encode('utf-8'))
+        file_name = page_name + '.html'
```

Comment:
separate to_html_stream and declare it as a dessia export

Reply:
Added a Figure object that inherits of PlotDataObject and concerns only plots (graph2d, Scatter, multiplot,...), to handle width and height in it. These attributes are probably useless in HTML, but i keep them for the moment, to avoid surprises.

plot_canvas function is now a function that calls the to_html method of plot_data_object that calls a to_html_stream method with lots of arguments. Can they stay here or should i find a way to only use stream argument ?

Is it the way you saw it ?